packages/meteor/asynchronous_queue.js

```diff
@@ -0,0 +1,154 @@
+class AsynchronousQueue {
+  /**
+   * Creates a queue that processes tasks in parallel batches while preserving completion order
+   * when needed. Configurable batch size and concurrency limits help optimize throughput.
+   */
+  constructor({
+                batchSize = 50,
+                maxConcurrent = 10,
+                orderMatters = false
+              } = {}) {
```

Comment:
All usages of  `Meteor._AsynchronousQueue` are with the default params; maybe those options could become env variables?